ducktape/utils/http_utils.py

```diff
@@ -25,4 +26,16 @@ def http_request(self, url, method, data="", headers=None):
 
         req = urllib2.Request(url, data, headers)
         req.get_method = lambda: method
-        return urllib2.urlopen(req)
\ No newline at end of file
+        try:
+            # The timeout parameter in urllib2.urlopen has strange behavior, and
+            # seems to raise errors when set to a number. Using an opener works however.
+            opener = urllib2.build_opener()
+            if timeout is None:
+                response = opener.open(req)
+            else:
+                response = opener.open(req, timeout=timeout)
+
+            return response
+
+        except Exception as e:
```

Comment:
Doesn't seem like this try/except is doing anything. Did you mean to translate it to a ducktape exception or should we just remove the block?